File: Php/app/Homegame/UniqueSlugValidator.php
namespace app\Homegame{

	use core\Validation\SimpleValidator;
	use Infrastructure\Data\Interfaces\HomegameStorage;

	class UniqueSlugValidator extends SimpleValidator {

		protected $homegameStorage;

		public function __construct($subject, $message, \Infrastructure\Data\Interfaces\HomegameStorage $homegameStorage){
			parent::__construct($subject, $message);
			homegameStorage = $homegameStorage;
		}

		public function validateSubject(){
			$existingHomegame = homegameStorage.getRawHomegameByName(subject);
			if($existingHomegame != null){
				addError(message);
				return false;
			}
			return true;
		}

	}

}

File: Web/Validators/UniqueSlugValidator.cs
using Infrastructure.Data.Storage.Interfaces;

namespace Web.Validators{

	public class UniqueSlugValidator : SimpleValidator
    {
	    private readonly IHomegameStorage _homegameStorage;

	    public UniqueSlugValidator(string subject, string message, IHomegameStorage homegameStorage)
            : base(subject, message)
		{
		    _homegameStorage = homegameStorage;
		}

	    protected override bool ValidateSubject(){
			var existingHomegame = _homegameStorage.GetHomegameByName(Subject);
			if(existingHomegame != null){
				AddError(Message);
				return false;
			}
			return true;
		}

	}

}